bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/menuitem/listener/LanguageItemSelectionListener.java

```diff
@@ -0,0 +1,63 @@
+/*******************************************************************************
+ * Copyright 2021-2022 Espressif Systems (Shanghai) PTE LTD. All rights reserved.
+ * Use is subject to license terms.
+ *******************************************************************************/
+
+package com.espressif.idf.ui.menuitem.listener;
+
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.widgets.MenuItem;
+import org.eclipse.ui.PlatformUI;
+
+import com.espressif.idf.core.IDFCorePlugin;
+import com.espressif.idf.core.util.EclipseIniUtil;
+
+public class LanguageItemSelectionListener extends SelectionAdapter
+{
+	private static final String LANGUAGE_SWITCH = "-nl";
+	
+	EclipseIniUtil eclipseIniUtil;
+
+	public LanguageItemSelectionListener()
+	{
+		try
+		{
+			eclipseIniUtil = new EclipseIniUtil();
+		}
+		catch (Exception e)
+		{
+			IDFCorePlugin.getPlugin().getLog().error(e.getMessage());
+		}
+	}
+
+	@Override
+	public void widgetSelected(SelectionEvent selectionEvent)
+	{
+		MenuItem currentItem = ((MenuItem) selectionEvent.widget);
+		String code = currentItem.getText().split("-")[1].replaceAll("^[ \t]+|[ \t]+$", "");
+		if (eclipseIniUtil != null)
+		{
+			try
+			{
+				if (currentItem.getText().toLowerCase().contains("default"))
+				{
+					if (eclipseIniUtil.containsEclipseSwitchInEclipseIni(LANGUAGE_SWITCH))
+					{
+						eclipseIniUtil.removeEclipseSwitch(LANGUAGE_SWITCH);
+					}
+				}
+				else
+				{
+					eclipseIniUtil.setEclipseSwitchValue(LANGUAGE_SWITCH, code);
+				}
```

Comment:
Hmm..why we are not using setVmArgInEclipseIni(LANGUAGE_SWITCH, code)? i think this what we want to update right

Reply:
I add this option for future use where we may need to pass arguments to the vm in eclipse any arguments passed after -vmargs in eclipse.ini are considered the arguments for the vm so our switch need to be before that

Reply:
Hi @alirana01 I understand your point but that's only going to increase technical debt in terms of code maintenance. I would suggest let's not keep any piece of code that is not used for now.

Reply:
https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it

Reply:
@kolipakakondal done